bionic/cache.py

```diff
@@ -961,20 +960,42 @@ def dependencies_exactly_match(self, prov):
 
 
 # Helpers for managing files.
-FILE_URL_PREFIX = 'file://'
+FILE_SCHEME = 'file'
+GCS_SCHEME = 'gs'
+SUPPORTED_SCHEMES = [FILE_SCHEME, GCS_SCHEME]
+
+
+class UnsupportedSchemeError(Exception):
+    pass
 
 
 def is_file_url(url):
-    return url.startswith(FILE_URL_PREFIX)
+    result = urlparse(url)
+    return result.scheme == FILE_SCHEME
+
+
+def is_gcs_url(url):
+    result = urlparse(url)
+    return result.scheme == GCS_SCHEME
+
+
+def is_absolute_url(url):
+    result = urlparse(url)
+    if not result.scheme:
+        return False
+    if result.scheme not in SUPPORTED_SCHEMES:
+        raise UnsupportedSchemeError(
+            f'Found a URL with unsupported scheme {result.scheme}.')
```

Comment:
BTW, if you use `{result.scheme!r}`, it will put quotes around the scheme (by applying `repr`), which may make the message less confusing in some circumstances.

`Found a URL with unsupported scheme data.`
vs
`Found a URL with unsupported scheme 'data'.`